jscomp/runtime/typed_array.ml

```diff
@@ -17,48 +17,329 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
+type array_buffer
+type 'a array_like (* should be shared with js_array *)
```

Comment:
ArrayBuffer actually belongs to the TypedArray spec. It's the underlying implementation for all the different typed array "views". Array-like is just a convention, really, but one that's very heavily relied on thoughout js apis, so might still make sense to make it a module.